presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -103,13 +104,14 @@ public LookupSourceFactory compileLookupSourceFactory(List<? extends Type> types
         }
     }
 
-    public PagesHashStrategyFactory compilePagesHashStrategyFactory(List<Type> types, List<Integer> joinChannels)
+    public PagesHashStrategyFactory compilePagesHashStrategyFactory(List<Type> types, List<Integer> joinChannels, Optional<JoinFilterFunction> joinFilterFunction)
```

Comment:
maybe it should be named `outputFilterFunction`? Couldn't it be easily used in aggregation since `PagesHashStrategy` is used there.